src/main/scala/cromwell/engine/SymbolStore.scala

```diff
@@ -23,7 +24,7 @@ object SymbolStoreEntry {
 case class SymbolStoreEntry(key: SymbolStoreKey, wdlType: WdlType, wdlValue: Option[WdlValue]) {
 
   wdlValue match {
-    case Some (value) => wdlType.checkCompatible(value.value)
+    case Some (value) => if(wdlType != value.wdlType) throw new UnsupportedOperationException("Fix this.")
```

Comment:
and by "implement checks to ensure that this won't happen" I mean checks at the AST/binding level.  We don't have those yet though.